controllers/agentaction_controller.go

```diff
@@ -401,7 +401,7 @@ func (r *AgentActionReconciler) createAgentJob(ctx context.Context, log logr.Log
 	var backoffLimit *int32
```

Comment:
I think this can be simplified a lot more to just setting the value on the spec directly using the value from the agent config.

```
porterJob.Spec.BackoffLimit = agentCfg.RetryLimit
```

I don't think we need the GetRetryLimit function, or the checks for existence, etc. If the user set the value to a non-nil value, the job will use the user-define limit. Otherwise nil will be assigned, and the k8s default (6) used instead.

Reply:
I have updated the code to not have the checks. However, the `GetRetryLimit` function is needed since the `AgentConfigSpecAdapter` is being used here instead of the raw `AgentConfigSpec`